test/functional/rpc_psbt.py

```diff
@@ -237,6 +237,43 @@ def test_sighash_mismatch(self):
 
         wallet.unloadwallet()
 
+    def test_sighash_adding(self):
+        self.log.info("Test adding of sighash type field")
+        self.nodes[0].createwallet("sighash_adding")
+        wallet = self.nodes[0].get_wallet_rpc("sighash_adding")
+        def_wallet = self.nodes[0].get_wallet_rpc(self.default_wallet_name)
+
+        addr = wallet.getnewaddress(address_type="bech32")
+        def_wallet.sendtoaddress(addr, 5)
+        self.generate(self.nodes[0], 6)
+
+        # Make a PSBT
+        psbt = wallet.walletcreatefundedpsbt([], [{def_wallet.getnewaddress(): 1}])["psbt"]
+        psbt = wallet.walletprocesspsbt(psbt=psbt, sighashtype="ALL|ANYONECANPAY", finalize=False)["psbt"]
+
+        # Check that the PSBT has a sighash field on all inputs
+        dec_psbt = self.nodes[0].decodepsbt(psbt)
+        for input in dec_psbt["inputs"]:
+            assert_equal(input["sighash"], "ALL|ANYONECANPAY")
```

Comment:
Atm, there is only 1 input in the PSBT. It would be nice to have another input that is Taproot and check for its sighash.

Reply:
Done